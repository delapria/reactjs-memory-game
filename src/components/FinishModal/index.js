import React from 'react';
import Modal from 'react-modal';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
    alignItems: 'center',
  },
};

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const TextName = styled.span`
  font-size: 15px;
  font-weight: bold;
`;

const ButtonNewPlayer = styled.button`
  margin-top: 10px;
  border-radius: 5px;
  background-color: #00c853;
  font-weight: bold;
  border: none;
  cursor: pointer;
  height: 30px;
`;

const ButtonResetGame = styled.button`
  border-radius: 5px;
  font-weight: bold;
  background-color: #f57f17;
  border: none;
  cursor: pointer;
  height: 30px;
`;

const FinishModal = ({
  modalIsOpen,
  setCloseModal,
  setResetGame,
  playerName,
  countPlays,
}) => {
  return (
    <div>
      <Modal
        isOpen={modalIsOpen}
        onRequestClose={setCloseModal}
        style={customStyles}
        contentLabel='Finish game'
        ariaHideApp={false}
      >
        <Container>
          <TextName>{`Jogador: ${playerName}`}</TextName>
          <h3>{`Jogadas: ${countPlays}`}</h3>
          <ButtonResetGame onClick={setResetGame}>
            Reiniciar o jogo
          </ButtonResetGame>
          <ButtonNewPlayer onClick={setCloseModal}>
            Novo jogador
          </ButtonNewPlayer>
        </Container>
      </Modal>
    </div>
  );
};

FinishModal.propTypes = {
  modalIsOpen: PropTypes.bool.isRequired,
  setCloseModal: PropTypes.func.isRequired,
  setResetGame: PropTypes.func.isRequired,
  playerName: PropTypes.string.isRequired,
  countPlays: PropTypes.number.isRequired,
};

export default FinishModal;
