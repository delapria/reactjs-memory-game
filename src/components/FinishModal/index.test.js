import React from 'react';
import renderer from 'react-test-renderer';
import Modal from 'react-modal';
import { shallow } from 'enzyme';

import FinishModal from '.';

describe('Snapshot Menu', () => {
  test('Testar se o componente renderiza de acordo com a snapshot', () => {
    const setCloseModal = () => {};
    const setResetGame = () => {};
    const modalIsOpen = false;
    const playerName = 'Douglas';
    const countPlays = 14;
    const treeJSON = renderer
      .create(
        <FinishModal
          modalIsOpen={modalIsOpen}
          playerName={playerName}
          countPlays={countPlays}
          setCloseModal={setCloseModal}
          setResetGame={setResetGame}
        />
      )
      .toJSON();
    expect(treeJSON).toMatchSnapshot();
  });
});

describe('Shallow tests', () => {
  test('Testa valor propriedade isOpen', () => {
    const setCloseModal = () => {};
    const setResetGame = () => {};
    const modalIsOpen = false;
    const playerName = 'Douglas';
    const countPlays = 14;
    const comp = shallow(
      <FinishModal
        modalIsOpen={modalIsOpen}
        playerName={playerName}
        countPlays={countPlays}
        setCloseModal={setCloseModal}
        setResetGame={setResetGame}
      />
    );
    expect(comp.getElement().props.children.props.isOpen).toBe(false);
  });
});
