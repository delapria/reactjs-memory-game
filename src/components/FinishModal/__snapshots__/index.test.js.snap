// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Menu Testar se o componente renderiza de acordo com a snapshot 1`] = `<div />`;
